### PR TITLE
feat(treesitter): add `#if?` predicate for conditional queries

### PR DESCRIPTION
**Pull request by clason**
### Problem 

There is no way to control queries at runtime, meaning controversial conceals or expensive injections have to be left to user config (and are not possible to conditionally enable). This is a regression over legacy syntax, which gates some features behind configuration variables.

### Solution

 Add an `#if?` predicate that can be used to gate patterns behind a configuration variable, where buffer variables take precedence. For example, the pattern
```lisp
((identifier) @foo (#if? "identifier-foo"))
```
is only applied if
1. `vim.b.identifier-foo` is truthy or
2. `vim.g.identifier-foo` is truthy and `vim.b.identifier-foo` is not falsy.

### Open question

 Should these variables be scoped, e.g., `vim.b.treesitter.identifier-foo`, or be part of a single dict (`vim.g.treesitter`)? If so, what color should that shed be?

### CODE REVIEW

**Comment by justinmk** [comment]
> Should these variables be scoped, e.g., `vim.b.treesitter.identifier-foo`

On the one way, constrainting to `b:treesitter` (`b:syntax`?) seems arbitrary since this `#if` mechanism looks similar to traditional syntax plugin mechanisms (but that's not necessarily a good thing: lack of constraints makes it it hard to optimize anything).

OTOH, constraining to `b:treesitter` is more of a 2-way door.

Will this `#if` predicate be able to request nested keys? E.g. can it check for `b:foo.bar`? If not, then that's another strong reason to define a subset (`b:treesitter`). Because otherwise `b:` gets polluted with lots of random top-level flags.

**Comment by clason** [comment]
>  E.g. can it check for b:foo.bar?

Not in the current, simple, implementation; it checks for a single variable. It's possible of course to split the string and then recursively test whether the nested dicts exist, but I'd prefer to keep this simple for performance reasons if possible.

**Comment by lewis6991** [comment]
As with all changes like this, I'm not a fan of anything that pushes us away from sharing queries with other applications.  Or at least I (personally) value that significantly more than being able to configure syntax highlighting in a way similar to `:help syn-file-remarks` as that's not something I've ever used.

One of the biggest advantages of treesitter is how agnostic it is, and we should actively avoid propagating any Nvim specific predicates and directives.

File: runtime/doc/treesitter.txt [hunk 1]
```diff
@@ -237,6 +237,14 @@ The following predicates are built in:
         This is the recommended way to check if the node matches one of many
         keywords, as it has been optimized for this.
 
+    `if?`                                            *treesitter-predicate-of?*
+        Apply the pattern if the specified variable is truthy. If the
+        corresponding buffer variable is not `nil`, use its value; otherwise
+        the global variable is checked). For example, the following pattern is
```
Comment (MariaSolOs) [inline runtime/doc/treesitter.txt:243]: Where is the opening parenthesis?
```suggestion
        the global variable is checked. For example, the following pattern is
```

Comment (clason) [reply runtime/doc/treesitter.txt:243]: It's on vacation.

**Pull request closed without merging**